staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -693,6 +693,8 @@ type APIResource struct {
 	Verbs Verbs `json:"verbs" protobuf:"bytes,4,opt,name=verbs"`
 	// shortNames is a list of suggested short names of the resource.
 	ShortNames []string `json:"shortNames,omitempty" protobuf:"bytes,5,rep,name=shortNames"`
+	// groupNames is a list of group aliases this resource belongs to (e.g. 'all')
+	GroupNames []string `json:"groupNames,omitempty" protobuf:"bytes,6,rep,name=groupNames"`
```

Comment:
> hmm, that seems backwards. can we reverse that before we ship a collision?

As I said, "while looking at writing a new kind of RESTMapper".  Fix for 1.7, but given layering probably not for 1.6.